PyDSS/pyControllers/pyControllerAbstract.py

```diff
@@ -1,9 +1,19 @@
+import abc
 
-class ControllerAbstract():
-    def __init__(self):
 
+class ControllerAbstract(abc.ABC):
+
+    def __init__(self, controlledObj, Settings, dssInstance, ElmObjectList, dssSolver):
```

Comment:
Just curious, why did you add this?  You aren't storing the parameters as instance variables.  You could do that and remove some minor redundancy.